### ISSUE TITLE
Error deploying coredns v1.9.0-eksbuild.7 in EKS Fargate

### ISSUE THREAD

**Issue by carolyn-mcclain** [gruntwork-io/kubergrunt#221]
**Describe the bug**
Deploying [coredns:v1.9.3-eksbuild.7](http://602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/coredns:v1.9.3-eksbuild.7) on fargate results in an error. 
```text
 Error: Error running command 'python '.terraform/modules/eks_cluster/modules/eks-cluster-control-plane/scripts/find_and_run_kubergrunt.py' -- eks schedule-coredns fargate --eks-cluster-name secopstn16-srv-us-east-1-eks --fargate-profile-arn arn:aws:eks:us-east-1:373333731569:fargateprofile/secopstn16-srv-us-east-1-eks/control-plane-services/28c68cc6-548f-48de-0a5f-51519c8ccbb1': exit status 1. Output: [] time="2024-01-17T15:31:16-07:00" level=info msg="No context name provided. Using default." name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="No kube config path provided. Using default (/home/comeaug/.kube/config)" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Got region us-east-1" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Got cluster arn arn:aws:eks:us-east-1:373333731569:cluster/secopstn16-srv-us-east-1-eks" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Doing fargate annotation" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Creating temporary file to act as kubeconfig with auth info" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Created /tmp/2104843637 to act as temporary kubeconfig file." name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Retrieving details for EKS cluster arn:aws:eks:us-east-1:373333731569:cluster/secopstn16-srv-us-east-1-eks" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Detected cluster deployed in region us-east-1" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Successfully retrieved EKS cluster details" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Appending cluster info for default to kubectl config." name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Successfully appended cluster info." name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Adding auth info to config" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Done adding auth info to config" name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Appending context default to kubectl config." name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Successfully appended context to kubectl config." name=kubergrunt
 [] time="2024-01-17T15:31:16-07:00" level=info msg="Running command: kubectl --kubeconfig /tmp/2104843637 patch deployment coredns -n kube-system --type json --patch [{\"op\": \"remove\",\"path\": \"/spec/template/metadata/annotations/eks.amazonaws.com~1compute-type\"}]"
 [] time="2024-01-17T15:31:17-07:00" level=info msg="The request is invalid"
 ERROR: exit status 1 
```

Module versions & Errant code: 
We are using: https://github.com/gruntwork-io/terraform-aws-eks/blob/v0.42.3/modules/eks-cluster-control-plane/main.tf. The errant command is on line 185.

The most current version https://github.com/gruntwork-io/terraform-aws-eks/blob/v0.65.4/modules/eks-cluster-control-plane/main.tf also uses the same command on line 260.

>   \# Use kubectl to remove the ec2 compute-type annotation from the coredns Deployment resource.
>   provisioner "local-exec" {
>     working_dir = path.module
>     command = join(
>       " ",
>       [
>         "python",
>         "./scripts/find_and_run_kubergrunt.py",
>         "--",
>         "eks", "schedule-coredns", "fargate",
>         "--eks-cluster-name", self.cluster_name,
>         "--fargate-profile-arn", self.arn,
>       ],
>     )
>   }
> 

**To Reproduce**
Attempt to deploy the eks cluster control plane with fargate enabled.

EKS version 1.25
kubergrunt versions tried v0.11.1 and v0.13.1

```hcl
module "eks_cluster" {
  source = "git::git@github.com:gruntwork-io/terraform-aws-eks.git//modules/eks-cluster-control-plane?ref=v0.42.3"

  cluster_name = var.cluster_name

  vpc_id                       = data.terraform_remote_state.vpc.outputs.vpc_id
  vpc_control_plane_subnet_ids = local.control_plane_subnet_ids

  enabled_cluster_log_types    = ["api", "audit", "authenticator"]
  kubernetes_version           = var.kubernetes_version
  endpoint_public_access       = var.endpoint_public_access
  endpoint_public_access_cidrs = var.endpoint_public_access_cidrs

  use_kubergrunt_verification = false
  use_upgrade_cluster_script  = false

  configure_openid_connect_provider          = var.configure_openid_connect_provider
  schedule_control_plane_services_on_fargate = true
  vpc_worker_subnet_ids                      = local.control_plane_subnet_ids

  fargate_profile_dependencies = [module.eks_k8s_role_mapping.aws_auth_config_map_name]
  custom_tags_eks_cluster      = var.custom_tags
  custom_tags_security_group   = var.custom_tags
}

```

**Expected behavior**
coredns to deploy on fargate without error

**Nice to have**
- [ ] Terminal output
- [ ] Screenshots

**Additional context**
The issue appears to be that the need to remove the ec2 compute type is no longer necessary as of Wednesday afternoon.
The AWS image being used for coredns did change between successful deployment on Tuesday:
   Image:      [602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/coredns:v1.8.7-eksbuild.3](http://602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/coredns:v1.8.7-eksbuild.3)
to
   Image:      [602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/coredns:v1.9.3-eksbuild.7](http://602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/coredns:v1.9.3-eksbuild.7)

**Comment by brandon-langley-mf** [comment]
The error occurs because CoreDNS does not have that annotation; I'm still digging to try to find out WHY it no longer has that annotation.

**Comment by brandon-langley-mf** [comment]
> The error occurs because CoreDNS does not have that annotation; I'm still digging to try to find out WHY it no longer has that annotation.

So I confirmed that this was the issue.   I reproduced the behavior using the raw kubectl command:

`kubectl patch deployment coredns -n kube-system --type json --patch '[{"op": "remove", "path": "/spec/template/metadata/annotations/eks.amazonaws.com~1compute-type"}]'`

I was able to then workaround it by running the following Kubectl command and then re-applying:

kubectl patch deployment coredns -n kube-system --type json --patch '[{"op": "add","path": "/spec/template/metadata/annotations","value": {"eks.amazonaws.com/compute-type": "ec2"}}]'

It looks like something changed, and that annotation is no longer present in coredns by default; I think it could also be happen due to bugs like https://github.com/gruntwork-io/terraform-aws-service-catalog/issues/1923 if Kubergrunt were to run twice.

TLDR there probably needs to be defensive code to prevent kubergrunt attempting to remove the annotation when it's not there.
